### PR TITLE
/tg/ mecha - Part 1

### PR DESCRIPTION
**Pull request by tigercat2000**
Start of porting the /tg/station mecha update.
This commit only cleans up the code and adds the proper dependencies for
the mecha.

 - Removes relative pathing from /datum/events
 - Updated Process_Spacemove() to use atom/movable/proc/get_spacemove_backup()
   - Basically just makes launching off of unanchored objects in space an OOP behaviour
 - Removed styling atrocities and relative pathing from most of the mecha files.
  - Notable exceptions:
   - code/game/mecha/mech_bay.dm
   - code/game/mecha/mech_fabricator.dm
   - code/game/mecha/mecha_construction_paths.dm
   - code/game/mecha/mecha_parts.dm
   - code/game/mecha/mecha_wreckage.dm
   - code/game/mecha/paintkits.dm
 - Removed dyndomove/dyndoattackby/dyndobulletact. No more icky dynamic calls. Replaced by specific code for the 3 modules that used it.
 - Refactored module cooldown. It now uses addtimer, and power usage is hooked directly into the cooldown calls.
 - Added atom/movable/proc/has_buckled_mobs(). Currently not that useful, but, necessary for porting the multi-buckling system from /tg/.
 - Split code/game/mecha/equipment/tools/tools.dm into multiple files.
 - Removed snowflake behaviour from exosuit drill. It now calls turf.drill_act(src).
  - Different from /tg/: Allows you to drill any wall/floor normally.
 - Drill no longer gibs mobs. It deals 80 organ damage instead.
 - Drill can be used to harvest dead mobs.
 - Removed all global_iterator systems from mecha and mecha equipment. Everything now uses object processing.
 - Mecha now have a turn sound variable and step sound variable. Updated
   all subtypes to use these instead of snowflake domove procs.
 - Removed mecha_do_after, mecha now uses the normal do_after.
 - Removed enter_after, same as above.
 - /obj/mecha/Process_Spacemove no longer strangely calls the user's
   spacemove.
 - /obj/mecha/return_pressure now uses return_air instead of copypasted
   cabin air detection.
 - Same for /obj/mecha/return_temperature
 - Added /obj/mecha/Exited. Basically, properly clears occupant refs,
   even if they teleport out/otherwise exit improperly.
 - Added hooks for mecha action buttons; Didn't implement them yet.
 - Moved mecha UI to code/game/mecha/mecha_topic.dm
 - Fixed turfs not updating atmos when ChangeTurf(/turf/space) is used.
 - Updated visible_message and audible_message. Both now use
   get_mobs_in_view() to ensure that mobs inside containers can see
   messages.
 - Removed /obj/item/mecha_parts/mecha_equipment/tool subtype. It had no
   use. Any subtypes are now just subtypes of
   /obj/item/mecha_parts/mecha_equipment.

### CODE REVIEW

**Comment by Fruerlund** [comment]
This means we can get action buttons to switch between modules installed in a mech? Yaya good!

File: code/game/mecha/equipment/tools/other_tools.dm [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/tesla_energy_relay/proc/get_charge()
+	if(equip_ready) //disabled
+		return
+	var/area/A = get_area(chassis)
+	var/pow_chan = get_power_channel(A)
+	if(pow_chan)
+		return 1000 //making magic
+
```
Comment (Crazylemon64) [inline code/game/mecha/equipment/tools/other_tools.dm]: Is this proc used?

File: code/game/mecha/equipment/tools/other_tools.dm [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+	..()
+	generator_init()
+
+/obj/item/mecha_parts/mecha_equipment/generator/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/item/mecha_parts/mecha_equipment/generator/proc/generator_init()
+	fuel_amount = 0
+
```
Comment (Crazylemon64) [inline code/game/mecha/equipment/tools/other_tools.dm]: Why is this necessary, if all it does is set `fuel_amount` to 0, which is its default value?

File: code/game/mecha/medical/odysseus.dm [hunk 3]
```diff
@@ -12,44 +12,35 @@
...
+		builtin_hud_user = 0
 
-		..()
-		return
+	. = ..()
 
-//TODO - Check documentation for client.eye and client.perspective...
-/obj/item/clothing/glasses/hud/health/mech
-	name = "Integrated Medical Hud"
-	HUDType = DATA_HUD_MEDICAL_ADVANCED
```
Comment (Crazylemon64) [inline code/game/mecha/medical/odysseus.dm]: Ding dong the witch is dead

**Comment by GeneralChaos81** [comment]
I've done work to port Mech_bay.dm in #5385

File: code/game/mecha/equipment/tools/other_tools.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/mecha/equipment/tools/other_tools.dm]: not yet i think
